### PR TITLE
Update elastic-agent-client and rename control proto package to cproto

### PR DESCRIPTION
**Pull request by michel-laterman**
## Proposed commit message

cherry-pick [the commit](https://github.com/elastic/beats/pull/39224) to update the elastic-agent-client to a more recent version. The PR was [reverted](https://github.com/elastic/beats/pull/39585) because the resulting binary failed to start due to a proto namespace collision.

Rename control proto package to cproto so it does not conflict with elastic-agent-client import

Upgrade integration test found that we had an error when upgrading from the latest 7.17.x snapshots.
Investigation found that while the binary could compile and all tests pass, attempting to run it would cause a panic when the imports executed their `init` functions as the protobuf namespaces collided.

## Checklist

- [x] My code follows the style guidelines of this project
- ~~I have commented my code, particularly in hard-to-understand areas~~
- ~~I have made corresponding changes to the documentation~~
- [ ] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## How to test this PR locally

compile the agent binary and attempt to run any command, i.e., `elastic-agent version`

## Related issues

- Relates https://github.com/elastic/elastic-agent/issues/4755

### CODE REVIEW

File: x-pack/elastic-agent/control.proto [hunk 1]
```diff
@@ -4,10 +4,10 @@
 
 syntax = "proto3";
 
-package proto;
+package cproto;
 
 option cc_enable_arenas = true;
-option go_package = "pkg/agent/control/proto;proto";
+option go_package = "pkg/agent/control/cproto;cproto";
```
Comment (cmacknz) [inline x-pack/elastic-agent/control.proto:10]: Can you still upgrade 7.17.x->7.17.x with this change?

In the 8.x agent we maintain two copies of the file but I can't remember if the concern there applies here as well. https://github.com/elastic/elastic-agent/blob/fd0d1bb164dd5dbd0f64c7a2d0e727cf7b13e777/control_v1.proto#L7-L11

**Review by belimawr — COMMENTED** [review]
LGTM, but I have the same question as Craig: Does it break any compatibility with older/newer versions that will need to communicate with this agent?

**Comment by michel-laterman** [comment]
I was able to upgrade successfully from 7.17.21-SNAPSHOT to 7.17.22-SNAPSHOT on a standalone install.
I was also able to upgrade afterwards to 8.15.0-SNAPSHOT on the same install.

**Review by belimawr — APPROVED** [review]

**Review by blakerouse — APPROVED** [review]
Looks good.

**Pull request merged**